inkcpp/value.cpp

```diff
@@ -303,6 +303,82 @@ namespace ink
 				inkFail("Invalid type for mod");
 			}
 
+			void convert(const char*& c, const data& d, char* number) {
+				switch(d.type) {
+				case data_type::int32:
+					snprintf(number, 32, "%d", d.integer_value);
+					break;
+				case data_type::uint32:
+					snprintf(number, 32, "%d", d.uint_value);
+					break;
+				case data_type::float32:
+					snprintf(number, 32, "%f", d.float_value);
+					break;
+				case data_type::newline:
+					number[0] = '\n';
+					number[1] = 0;
+					break;
+				case data_type::string_table_pointer:
+				case data_type::allocated_string_pointer:
+					c = d.string_val;
+					break;
+				default: number[0] = 0;
+				}
+			}
+			bool value::compare_string(const value& left, const value& right) {
+				size_t l_i = 0;
```

Comment:
This function looks great! Thanks for solving this long untouched problem of string comparison. If you could add a comment above each if/else/while that'd be great for readability.